pilot/cmd/pilot-agent/main.go

```diff
@@ -99,13 +99,18 @@ var (
 
 	wg sync.WaitGroup
 
-	instanceIPVar        = env.RegisterStringVar("INSTANCE_IP", "", "")
-	podNameVar           = env.RegisterStringVar("POD_NAME", "", "")
-	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
-	istioNamespaceVar    = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
-	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
-	sdsEnabledVar        = env.RegisterBoolVar("SDS_ENABLED", false, "")
-	sdsUdsPathVar        = env.RegisterStringVar("SDS_UDS_PATH", "unix:/var/run/sds/uds_path", "SDS address")
+	instanceIPVar                       = env.RegisterStringVar("INSTANCE_IP", "", "")
+	podNameVar                          = env.RegisterStringVar("POD_NAME", "", "")
+	podNamespaceVar                     = env.RegisterStringVar("POD_NAMESPACE", "", "")
+	istioNamespaceVar                   = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
+	kubeAppProberNameVar                = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	sdsEnabledVar                       = env.RegisterBoolVar("SDS_ENABLED", false, "")
+	sdsUdsPathVar                       = env.RegisterStringVar("SDS_UDS_PATH", "unix:/var/run/sds/uds_path", "SDS address")
+	stackdriverEnabled                  = env.RegisgterBoolVar("STACKDRIVER_ENABLED", false, "If enabled, stackdriver will get configured as the tracer.")
+	stackdriverDebug                    = env.RegisgterBoolVar("STACKDRIVER_DEBUG", false, "")
```

Comment:
I think you can just read directly from mesh config, no need to copy all of this to environment variable

Reply:
This is needed to fill up the mesh config for stackdriver.. please see changes below..
doing this using environment variable and not using command line, so that command line params don't explode
Please Refer :https://github.com/istio/istio/issues/15117#issuecomment-540697121

Reply:
ok actually lgtm, I misunderstood

Reply:
Same here.. I was confused too initially :)